pkg/manifests/template/helm.go

```diff
@@ -170,19 +178,75 @@ func (h *helm) Render(svc *console.ServiceDeploymentForAgent, utilFactory util.F
 	return manifests, nil
 }
 
+func (h *helm) luaValues(svc *console.ServiceDeploymentForAgent) (map[string]interface{}, error) {
+	// Initialize empty results
+	newValues := make(map[string]interface{})
+	var valuesFiles []string
+
+	if svc == nil {
+		return nil, fmt.Errorf("no service found")
+	}
+	if svc.Helm == nil {
+		return newValues, nil
+	}
+	if svc.Helm.LuaScript == nil {
+		return newValues, nil
+	}
+	p := luautils.NewProcessor(h.dir)
+	defer p.L.Close()
+
+	// Register global values and valuesFiles in Lua
+	valuesTable := p.L.NewTable()
+	p.L.SetGlobal("values", valuesTable)
+
+	valuesFilesTable := p.L.NewTable()
+	p.L.SetGlobal("valuesFiles", valuesFilesTable)
+
+	// Execute the Lua script
+	err := p.L.DoString(*svc.Helm.LuaScript)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := luautils.MapLua(p.L.GetGlobal("values").(*lua.LTable), &newValues); err != nil {
+		return nil, err
+	}
+
+	if err := luautils.MapLua(p.L.GetGlobal("valuesFiles").(*lua.LTable), &valuesFiles); err != nil {
+		return nil, err
+	}
+
+	for _, file := range valuesFiles {
+		currentMap, err := h.valuesFile(svc, file)
+		if err == nil {
+			newValues = algorithms.Merge(newValues, currentMap)
+		}
+	}
+
+	return newValues, nil
+}
+
 func (h *helm) values(svc *console.ServiceDeploymentForAgent) (map[string]interface{}, error) {
 	currentMap, err := h.valuesFile(svc, "values.yaml.liquid")
 	if err != nil {
 		return currentMap, err
 	}
-	if svc.Helm != nil && svc.Helm.ValuesFiles != nil && len(svc.Helm.ValuesFiles) > 0 {
+	if svc.Helm != nil {
 		for _, f := range svc.Helm.ValuesFiles {
```

Comment:
what happens if ValuesFiles is nil here?

Reply:
Seems like we should still keep that condition even if it's theoretically covered elsewhere, just in case

Reply:
in Go, range safely handles nil for slices

Reply:
there is no need to double-check it: https://go.dev/play/p/d4h5n4pfSn7
and the `svc.Helm.ValuesFiles` is used only in this place